search/search_repository.go

```diff
@@ -370,11 +370,18 @@ func (q Query) generateExpression() (criteria.Expression, error) {
 			return nil, errors.NewBadParameterError("key not found", q.Name)
 		}
 		left := criteria.Field(key)
-		right := q.determineLiteralType(key, *q.Value)
-		if q.Negate {
-			myexpr = append(myexpr, criteria.Not(left, right))
+		if q.Value != nil {
+			right := q.determineLiteralType(key, *q.Value)
+			if q.Negate {
+				myexpr = append(myexpr, criteria.Not(left, right))
+			} else {
```

Comment:
you can get rid of this else statement

Reply:
But it will append `Equals` too when Negate is `true`